src/RawTransaction.php

```diff
@@ -980,6 +980,7 @@ public static function validate_signed_transaction($raw_tx, $json_string, $magic
                     foreach ($redeemScript['keys'] as $public_key) {
                         if (self::_check_sig($signature, $message_hash[$i], $public_key) == true) {
                             $pubkey_found = true;
```

Comment:
For P2SH, all this does is ensure for each key in the script, if there's a signature it should be valid. That said, it doesn't return false if one of the signatures is actually invalid if you have more than one signature.. 

Perhaps this could be refactored, as it shouldn't do m^2 signature checks, but rather only m signature checks, since signatures must come in the order of the redeemScript. (This probably isn't being enforced in the library).